### ISSUE TITLE
Not working with require.js

### ISSUE THREAD

**Issue by thany** [eligrey/FileSaver.js#383]
When I include file-saver with requirejs, it always exports as `undefined`.

My requirejs config:
```javascript
{
    baseUrl: '.',
    paths: {
      // ...
      "FileSaver": "vendor/file-saver/FileSaver"
    }
}
```

I can see the browser successfully loads the file FileSaver.js, and executes it absolutely fine.

But then I try using it in a module:
```javascript
define(["angular", "FileSaver"], function (angular, fileSaver) {
  "use strict";

  console.log(fileSaver);
});
```

And it logs `undefined`.

This is likely because of how file-saver defines itself:
```javascript
define("FileSaver.js", function() {
  return saveAs;
});
```

It should be:
```javascript
define([ ], function() {
  return saveAs;
});
```

And then it'll work.

This is because the module name ends in `.js`. Requirejs will assume a module name like that to be a *file* instead of a defined module, and will thus try loading `/FileSaver.js` which obviously won't exist.

**Comment by ogonkov** [comment]
Still broken for R.JS (works OK for Webpack)

Also, exports for AMD and CJS are different

For CJS module exports object w/ `saveAs` method, but for AMD it exports as method itself.

**Comment by eligrey** [comment]
I'm willing to merge any pull request that fixes this without breaking other integrations.

**Comment by eligrey** [comment]
Fixed by #391
